src/modules/src/Eryph.Modules.VmHostAgent/Networks/ArpUpdateRequestedEventHandler.cs

```diff
@@ -0,0 +1,72 @@
+﻿using System;
+using System.Linq;
+using System.Net;
+using System.Threading.Tasks;
+using Eryph.Messages.Resources.Catlets.Events;
+using JetBrains.Annotations;
+using LanguageExt;
+using Microsoft.Extensions.Logging;
+using Rebus.Handlers;
+using Vanara.PInvoke;
+
+namespace Eryph.Modules.VmHostAgent.Networks;
+
+[UsedImplicitly]
+public class ArpUpdateRequestedEventHandler(ILogger log, IWindowsArpCache arpCache) : IHandleMessages<ArpUpdateRequestedEvent>
+{
+    public Task Handle(ArpUpdateRequestedEvent message)
+    {
+        log.LogTrace("Entering {method}", nameof(ArpUpdateRequestedEventHandler));
+
+        var sw = System.Diagnostics.Stopwatch.StartNew();
+
+        var addressBytes =
+            message.UpdatedAddresses
+                .Select(a => IPAddress.Parse(a.IpAddress).GetAddressBytes()).ToArray();
+
+        // in case ARP table is very large, we can optimize this by using a lookup table
+        // that is already filtered by dynamic entries and the addresses we are interested in
+        var arpTable = arpCache.GetIpNetTable();
```

Comment:
There are Powershell Cmdlet which support this as well: `Get-NetNeighbor`, `Remove-NetNeighbor`.
Not sure if it makes sense to use them (performance, etc.) but they seem to support IPv4 and
IPv6 with the same code path.